cinema/views.py

```diff
@@ -61,71 +54,21 @@ class CinemaHallViewSet(
     permission_classes = (IsAdminOrIfAuthenticatedReadOnly,)
 
 
-class MovieViewSet(
-    mixins.ListModelMixin,
-    mixins.CreateModelMixin,
-    mixins.RetrieveModelMixin,
-    viewsets.GenericViewSet,
-):
-    queryset = Movie.objects.prefetch_related("genres", "actors")
+class MovieViewSet(viewsets.ModelViewSet):
+    queryset = Movie.objects.all()
     serializer_class = MovieSerializer
-    authentication_classes = (TokenAuthentication,)
-    permission_classes = (IsAdminOrIfAuthenticatedReadOnly,)
-
-    @staticmethod
-    def _params_to_ints(qs):
-        """Converts a list of string IDs to a list of integers"""
-        return [int(str_id) for str_id in qs.split(",")]
-
-    def get_queryset(self):
-        """Retrieve the movies with filters"""
-        title = self.request.query_params.get("title")
-        genres = self.request.query_params.get("genres")
-        actors = self.request.query_params.get("actors")
-
-        queryset = self.queryset
-
-        if title:
-            queryset = queryset.filter(title__icontains=title)
 
-        if genres:
-            genres_ids = self._params_to_ints(genres)
-            queryset = queryset.filter(genres__id__in=genres_ids)
-
-        if actors:
-            actors_ids = self._params_to_ints(actors)
-            queryset = queryset.filter(actors__id__in=actors_ids)
-
-        return queryset.distinct()
-
-    def get_serializer_class(self):
-        if self.action == "list":
-            return MovieListSerializer
-
-        if self.action == "retrieve":
-            return MovieDetailSerializer
-
-        if self.action == "upload_image":
-            return MovieImageSerializer
-
-        return MovieSerializer
-
-    @action(
-        methods=["POST"],
-        detail=True,
-        url_path="upload-image",
-        permission_classes=[IsAdminUser],
+    @extend_schema(
+        parameters=[
+            OpenApiParameter("title", type=str, description="Filter by movie title"),
+            OpenApiParameter("genres", type=str, description="Filter by genres"),
+            OpenApiParameter("actors", type=str, description="Filter by actors"),
+            OpenApiParameter("date", type=str, description="Filter by session date"),
+            OpenApiParameter("movie", type=str, description="Filter by movie in session"),
```

Comment:
The parameter 'movie' in the OpenApiParameter might be confusing since it overlaps with the 'movie' parameter used in the MovieSessionViewSet's get_queryset method. Consider renaming it to something more descriptive, like 'movie_id', to avoid confusion.